tools/ingest-converter/src/main/resources/ingest-shared.js

```diff
@@ -109,12 +109,38 @@ var IngestConverter = {
             ? this.quote_string(this.dots_to_square_brackets(patterns[0]))
             : this.create_pattern_array(patterns);
     },
-    
+
     filter_hash: function(contents) {
         return this.fix_indent(this.create_hash("filter", contents))
     },
-    
+
     filters_to_file: function(filters) {
         return filters.join("\n\n") + "\n";
+    },
+
+    /**
+     * Does it have an on_failure field?
+     * @param processor Json
+     * @param name Name of the processor
+     * @returns {boolean} True if has on failure
+     */
+    has_on_failure: function (processor, name) {
```

Comment:
Yep. The plan is to use this in other processors such as geoip (which produces a failure tag called `_geoip_lookup_failure`), and others as well..

Only implemented for grok now though.